### ISSUE TITLE
Implement basic functionality

### ISSUE THREAD

**Issue by zoidyzoidzoid** [yola/teamsupport-python#1]
[API docs](http://help.teamsupport.com/1/en/topic/api)
# Basic methods needed

| URL | method | use |
| --- | --- | --- |
| `Tickets` | GET | Lists all of the tickets without custom fields |
| `Tickets` | POST | Creates a new ticket. (Uses custom fields) |
| `Tickets/{TicketID}` | GET | Gets a single ticket |
| `Tickets/{TicketID}` | PUT | Updates an existing ticket |
| `Tickets/{TicketID}` | DELETE | Deletes an existing ticket |
| `Tickets/{TicketID}/Actions` | GET | Lists actions for a ticket |
| `Tickets/{TicketNumber}/Actions/{ActionID}` | GET | Gets a single action |
| `Tickets/{TicketNumber}/Actions/{ActionID}` | PUT | Updates an existing action |

**Comment by zoidyzoidzoid** [comment]
## Update

I'm working on this. For a day or two I wasn't sure if I should be so I was hesitant. Trying to make it as clean as possible, which is tricky with XML. Trying to take as much as I can from our existing LiveLookup integration and factor out as much repetitive code as I can.

Waiting on all of our custom fields to be added, but I've got an API account and token now, with which I can start running tests. I've implemented all four methods, just gotta test integration to make sure the XML data is formatted properly, since we had issues with HelpSpot since data needed to be encoded as `iso-8859-1`.

**Comment by MarijeP** [comment]
@zoidbergwill I've added the custom field stuff to https://github.com/yola/p-support/issues/73. I'm not sure on the types etc, so if there's iffyness, let me know and I can fiddle.

I'm not sure on syncing data like name and email address. Let me know if you're looking for that info?

**Comment by zoidyzoidzoid** [comment]
## Update

I haven't had a lot of success, which has been slowing down development quite a lot. The documentation isn't great and the error responses from the API aren't helpful at all. The API just gives an HTML response with an `http 500`, no matter what you do wrong.

I've got integration tests working now that test normal ticket flow. I'm aiming at having working unit and integration tests by the beginning of next week.

**Comment by MarijeP** [comment]
Can we get you some help from them? What can I do to help?

**Comment by zoidyzoidzoid** [comment]
[Pull open](https://github.com/yola/teamsupport-python/pull/3)
